### PR TITLE
REFACTOR: Remove deprecated `action="something"` attributes

### PR DESCRIPTION
**Pull request by eviltrout**
In newer versions of Ember, `this.sendAction` is deprecated:

https://deprecations.emberjs.com/v3.x/#toc_ember-component-send-action

This patch removes most of our current uses (via `d-button`) but also
some other rogue `this.sendAction` calls too.

### CODE REVIEW

File: app/assets/javascripts/discourse/app/components/discourse-linked-text.js [hunk 1]
```diff
@@ -5,10 +5,10 @@ import discourseComputed from "discourse-common/utils/decorators";
 export default Component.extend({
   tagName: "span",
 
-  @discourseComputed("text")
+  @discourseComputed("text", "textParams")
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/app/components/discourse-linked-text.js:8]: this dosnt look related to actions ?

File: app/assets/javascripts/discourse/app/templates/components/pwa-install-banner.hbs [hunk 2]
```diff
@@ -4,12 +4,13 @@
       <span>
         {{discourse-linked-text
           action=(action "turnOn")
-          translatedText=(i18n "pwa.install_banner" title=siteSettings.title)
+          text="pwa.install_banner"
+          textParams=(hash title=siteSettings.title)
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/app/templates/components/pwa-install-banner.hbs:8]: doesn't look related to actions

Comment (eviltrout) [reply app/assets/javascripts/discourse/app/templates/components/pwa-install-banner.hbs:8]: my bad, while fixing the actions I found this overwritten CP so I fixed that too. I'll split it out.

**Comment by eviltrout** [comment]
@jjaffeux fixed

**Review by jjaffeux — APPROVED** [review]

File: app/assets/javascripts/discourse/app/templates/components/share-popup.hbs [hunk 3]
```diff
@@ -34,6 +34,11 @@
       </div>
     {{/if}}
 
-    {{d-button action="close" class="btn btn-flat close" icon="times" aria-label="share.close" title="share.close"}}
+    {{d-button action=(action "close")
```
Comment (CvX) [inline app/assets/javascripts/discourse/app/templates/components/share-popup.hbs:37]:
```suggestion
    {{d-button
      action=(action "close")
```

**Review by CvX — APPROVED** [review]
Added a small formatting suggestion.

On another note, we should be gradually moving away from `{{action}}` helper. See: https://guides.emberjs.com/v3.21.0/upgrading/current-edition/action-on-and-fn/

File: app/assets/javascripts/discourse/app/templates/components/share-popup.hbs (hunk 3 above)
Comment (jjaffeux) [reply app/assets/javascripts/discourse/app/templates/components/share-popup.hbs:37]: yes agreed, I didn't want to nitpick, but 😅 

Hope we can have auto formatting of templates soon.

**Comment by jjaffeux** [comment]
> On another note, we should be gradually moving away from `{{action}}` helper. See: https://guides.emberjs.com/v3.21.0/upgrading/current-edition/action-on-and-fn/

Also agreed, on this, I wonder if we shouldn't do some priority list of things to upgrade before and after ember-cli move.
- what can we do today?
- what should we do today?
- what should we do as soon as possible?
- what can wait?

**Comment by eviltrout** [comment]
So far I've been focused on deprecations in the Ember 3.15 branch I'm running on Ember CLI. Right now we have very few which feels great!
